src/runtime/hexagon_host.cpp

```diff
@@ -432,23 +444,26 @@ WEAK int halide_hexagon_run(void *user_context,
     int input_buffer_count = map_arguments(user_context, arg_count, arg_sizes, args, arg_flags, 0x3, 0x1,
                                            input_buffers);
     if (input_buffer_count < 0) {
-        return input_buffer_count;
+        error(user_context) << "Hexagon: map_arguments() failed.";
+        return halide_error_code_generic_error;
     }
 
     // Then the output buffers (bit 1 of flags is set).
     remote_buffer *output_buffers = input_buffers + input_buffer_count;
     int output_buffer_count = map_arguments(user_context, arg_count, arg_sizes, args, arg_flags, 0x2, 0x2,
```

Comment:
map_arguments notionally returns an error code, so it will have already called halide_error (though it looks like it can't actually throw an error).

Reply:
Sure, but the fundamental issue here is that with our current setup, there isn't a good way to detect that other than manual checking, and it's super-fragile (no obvious way to keep this in sync). With the status quo, calling error() too much is better than calling it not enough (since the latter risks dropping errors on the floor).

(FWIW, all this runtime churn is basically lead-up to a proposal for cleaning up the protocol in a way that should make this more tractable)

Reply:
I thought our new rule was that if something returns a halide_error_code_t, then it has already called halide_error if necessary.

Reply:
Also, if something returns a halide_error_code_t, return that error code upwards, don't rewrite it as the code here is doing.

Reply:
(I'm reading map_arguments as intending to return an error code if the return value is negative)

Reply:
> I thought our new rule was that if something returns a halide_error_code_t, then it has already called halide_error if necessary.

That was a rule that I had proposed, but I decided to abandon it (for now) because 
   (1) relying on that rule in a reliable way really needs us to convert all the `int` return types to `halide_error_code_t`, so that the person writing/reading the call site doesn't have to go inspect the callee to be satisfied that the return value really is a `halide_error_code_t`; I am very much in favor of making such a change, but it's quite invasive and making the change would require all downstream users to tweak runtime overrides to match, and while that's likely to be an easy change in almost all cases, it's a change we'd need to discuss more widely and get consensus on
    (2) I think there is a simpler approach we can use, which I haven't yet proposed here (basically, I've been harvesting IMHO-useful changes from a grand experiment and trying to get them landed first)
    
> Also, if something returns a halide_error_code_t, return that error code upwards, don't rewrite it as the code here is doing.

Fun fact: `map_arguments() doesn't return a `halide_error_code_t` at all, it returns an actual count! (See what I said above about having to inspect the callee? Yikes.) 

Another fun fact: `map_arguments()` cannot ever return a negative value, so this clause was never firing in the first place -- it really should be removed entirely.

> (I'm reading map_arguments as intending to return an error code if the return value is negative)

That's what the comment for it says, but (1) it doesn't actually do that, and (2) IMHO it's usually a false economy to handle errors in this way (ie, by using magic values of the return result to signal an error). Returning `nullptr` as failure from thing-returning-pointer is arguably reasonable (since `nullptr` is never a valid ptr), but returning negative values from an int result is tempting fate for a misinterpretation of the results someday. (Obviously, if it could be shown that there was meaningful codesize/performance savings by doing such a thing, it would be worth entertaining, but I will bet you a dollar that that isn't the case here.)